### ISSUE TITLE
The Carousel items do not render on 1st load.

### ISSUE THREAD

**Issue by s-pcode** [Learus/react-material-ui-carousel#133]
The carousel items are not visible on load but when I resize the screen or change code then on re-rendering it is visible.

Tech Stack-
1. NextJS-"^11.1.2",
2. MUI-"^5.0.0",
3. "react-material-ui-carousel": "^3.1.0",

The same issue remains even if I build and start the app. Please find my code below.

```jsx
import { motion } from "framer-motion";
// material
import { styled, alpha } from "@mui/material/styles";
import {
  Box,
  Container,
  Typography,
  useMediaQuery,
  useTheme,
} from "@mui/material";
// hooks
import Prev from "@mui/icons-material/KeyboardArrowLeftRounded";
import Next from "@mui/icons-material/KeyboardArrowRightRounded";
import { MotionInView, varFadeInUp, varFadeInDown } from "../animate";
import Carousel from "react-material-ui-carousel";

// ----------------------------------------------------------------------

const RootStyle = styled("div")(({ theme }) => ({
  padding: theme.spacing(10, 0, 0, 0),
  backgroundImage:
    theme.palette.mode === "light"
      ? `linear-gradient(180deg, ${theme.palette.grey[200]} 0%, ${alpha(
          theme.palette.grey[200],
          0
        )} 100%)`
      : "none",
}));

// ----------------------------------------------------------------------

export default function FeatureCarousel() {
  const isMobile = useMediaQuery(useTheme().breakpoints.down("md"));
  return (
    <RootStyle>
      <Container
        maxWidth="lg"
        sx={{ position: "relative", textAlign: "center" }}
      >
        <MotionInView variants={varFadeInUp}>
          <Typography variant="h2" sx={{ mb: 3, color: "#555555" }}>
            Deep Insights
          </Typography>
        </MotionInView>

        <Carousel
          navButtonsProps={{
            style: {
              backgroundColor: "#008080",
            },
          }}
          indicatorIconButtonProps={{
            style: {
              color: "grey",
            },
          }}
          activeIndicatorIconButtonProps={{
            style: {
              color: "#008080",
              backgroundColor: "#008080",
            },
          }}
          PrevIcon={<Prev />}
          NextIcon={<Next />}
          autoPlay
          duration={1000}
          navButtonsAlwaysVisible={isMobile ? false : true}
          stopAutoPlayOnHover
          indicators={isMobile ? true : false}
        >
          <Component1 />
          <Component2 />
          <Component3 />
        </Carousel>
      </Container>
    </RootStyle>
  );
}

function Component1() {
  return (
    <Box sx={{ position: "relative" }}>
      <Box component="img" src="/static/landing/grid.png" />

      <Box sx={{ position: "absolute", top: 0 }}>
        <MotionInView variants={varFadeInUp}>
          <img alt="screen" src={"/static/landing/portfolio/Dashboard.png"} />
        </MotionInView>
      </Box>

      <Box sx={{ position: "absolute", top: 0 }}>
        <MotionInView variants={varFadeInDown}>
          <motion.div
            animate={{ y: [-25, 5, -25] }}
            transition={{ duration: 8, repeat: Infinity }}
          >
            <img alt="sidebar" src={"/static/landing/portfolio/DM.png"} />
          </motion.div>
        </MotionInView>
      </Box>

      <Box sx={{ position: "absolute", top: 0 }}>
        <MotionInView variants={varFadeInDown}>
          <motion.div
            animate={{ y: [-5, 10, -5] }}
            transition={{ duration: 8, repeat: Infinity }}
          >
            <img alt="sidebar" src={"/static/landing/portfolio/NW.png"} />
          </motion.div>
        </MotionInView>
      </Box>

      <Box sx={{ position: "absolute", top: 0 }}>
        <MotionInView variants={varFadeInDown}>
          <motion.div
            animate={{ y: [-15, 5, -15] }}
            transition={{ duration: 10, repeat: Infinity }}
          >
            <img alt="sidebar" src={"/static/landing/portfolio/AD.png"} />
          </motion.div>
        </MotionInView>
      </Box>
    </Box>
  );
}

function Component2() {
  return (
    <Box sx={{ position: "relative" }}>
      <Box component="img" src="/static/landing/grid.png" />

      <Box sx={{ position: "absolute", top: 0 }}>
        <MotionInView variants={varFadeInUp}>
          <img alt="screen" src={"/static/landing/defi/Defi.png"} />
        </MotionInView>
      </Box>
      <Box sx={{ position: "absolute", top: 0 }}>
        <MotionInView variants={varFadeInDown}>
          <motion.div
            animate={{ y: [-15, 5, -15] }}
            transition={{ duration: 8, repeat: Infinity }}
          >
            <img alt="sidebar" src={"/static/landing/defi/card.png"} />
          </motion.div>
        </MotionInView>
      </Box>
    </Box>
  );
}

function Component3() {
  return (
    <Box sx={{ position: "relative" }}>
      <Box component="img" src="/static/landing/grid.png" />

      <Box sx={{ position: "absolute", top: 0 }}>
        <MotionInView variants={varFadeInUp}>
          <img alt="screen" src={"/static/landing/history/HSTRY.png"} />
        </MotionInView>
      </Box>

      <Box sx={{ position: "absolute", top: 0 }}>
        <MotionInView variants={varFadeInDown}>
          <motion.div
            animate={{ y: [-25, 5, -25] }}
            transition={{ duration: 8, repeat: Infinity }}
          >
            <img alt="sidebar" src={"/static/landing/history/HSTRY2.png"} />
          </motion.div>
        </MotionInView>
      </Box>

      <Box sx={{ position: "absolute", top: 0 }}>
        <MotionInView variants={varFadeInDown}>
          <motion.div
            animate={{ y: [-5, 10, -5] }}
            transition={{ duration: 8, repeat: Infinity }}
          >
            <img alt="sidebar" src={"/static/landing/history/HSTRY4.png"} />
          </motion.div>
        </MotionInView>
      </Box>

      <Box sx={{ position: "absolute", top: 0 }}>
        <MotionInView variants={varFadeInDown}>
          <motion.div
            animate={{ y: [-25, 5, -25] }}
            transition={{ duration: 10, repeat: Infinity }}
          >
            <img alt="sidebar" src={"/static/landing/history/HSTRY5.png"} />
          </motion.div>
        </MotionInView>
      </Box>
    </Box>
  );
}

```

Need help urgently. Thanks in advance 👍

**Comment by s-pcode** [comment]
Update:
I just figured that it does not work if the carousel items are images. How to add images into the carousel, please help.

**Comment by jessejamesrich** [comment]
I have the same problem. No matter what size attributes I put on what, it does not render? Any fixes or ideas about this? MUI 5.

**Comment by Learus** [comment]
@s-pcode 
Check out this sandbox [here](https://codesandbox.io/s/next-js-forked-ix0d4?file=/index.js)

I copy-pasted your code and added some generic images for it to work. Moreover, I commented out the parts that the `MotionInView` components are used since I didn't have access to them.

There doesn't seem to be a problem with not rendering on load in that sandbox. I urge you to change and play around with the sandbox in order to figure this out.

Maybe it has to do with exactly those `MotionInView` components and not with the Carousel or Next.

**Comment by clepixie** [comment]
Currently trying to troubleshoot the same issue, so I thought I would pop in with details of my own case.
Unfortunately installing `npm i react-material-ui-carousel@2.3.7-mui5` as how @jessejamesrich suggested did not resolve the 1st load render issue on my end. The only way I can get the Carousel to load on MUI 5.3.1 is with the Carousel with the most recent v5 support, but as Jesse said, it seems to reset the theme. I have stripped my carousel component code down to the test component below, but the rendering issue persists. Appreciate the help in advance.

**Problem:**
react-material-ui-carousel@2.3.7-mui5: In production mode, Carousel component only renders on re-renders; first render/load does not show component. 
react-material-ui-carousel@2.3.8: In production mode, Carousel component does render on first render, but breaks MUI theming like color palettes, buttons, etc.

**Tech Stack:**
next@12.0.7
@mui/material@5.3.1
@mui/icons-material@5.3.1
@mui/styles@5.3.0
react-material-ui-carousel@2.3.7-mui5

**Stripped code:**
```JS
import { Card } from "@mui/material"
import React from "react";
import Carousel from 'react-material-ui-carousel'

const TestCarousel = props => {

    return (
        <Carousel>
            <Card>
                Hello
            </Card>
            <Card>
                World
            </Card>
        </Carousel>
    )
}

export { TestCarousel }
```

**Comment by Learus** [comment]
This issue is fixed in v3.3.3. Tested with both Nextjs and CRA.

The carousel's height now dynamically changes to be exactly the currently active child's height. You can override this behavior by specifying the height prop.
